config_gnome2.c

```diff
@@ -34,23 +34,29 @@ typedef struct g_proxy_config_gnome2_s {
 
 g_proxy_config_gnome2_s g_proxy_config_gnome2;
 
-bool proxy_config_gnome2_get_auto_discover(void) {
-    char *mode = NULL;
-    bool auto_discover = false;
+static bool proxy_config_gnome2_is_mode(const char *mode) {
+    bool equal = false;
 
-    mode =
+    char *compare_mode =
```

Comment:
IMHO `compare_mode` is not a convenient. I would use `set_mode`. Or woild swap names at lease, `compare_mode` to the parameter name.

Reply:
Fixed.